### ISSUE TITLE
gpflow.utilities.freeze() fails in clear_bijector 

### ISSUE THREAD

**Issue by junhaobearxiong** [GPflow/GPflow#1556]
# Bug / performance issue / build issue

I'm fairly new to gpflow, so if I'm making a silly mistake here I apologize in advance :) but I'm trying to save a model's parameters by doing gpflow.utilities.freeze. The function seems to fail when clearing the bijectors.

## To reproduce

**Minimal, reproducible example**

```python
# This is the place for your code that reproduces the bug.
# Please make sure it does not depend on external libraries (beyond GPflow's own requirements) or specific datasets, and the smaller, the better :)
# For help on how to write a good bug report, see https://stackoverflow.com/help/minimal-reproducible-example
import gpflow

kernel = gpflow.kernels.SquaredExponential()
N = 10
M = 5  # Number of inducing locations
X = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]  # X values
Z = X[:M].copy()  # Initialize inducing locations to the first M inputs in the dataset

m = gpflow.models.SVGP(kernel, gpflow.likelihoods.Gaussian(), Z, num_data=N)
gpflow.utilities.freeze(m)
```

**Stack trace, or error message**
```
// Paste the full stack trace/error message here
Traceback (most recent call last):
  File "savemodel_error.py", line 10, in <module>
    gpflow.utilities.freeze(m)
  File "/Users/jxiong/Documents/Projects/ipsc_gplvm/env/lib/python3.6/site-packages/gpflow/utilities/utilities.py", line 281, in freeze
    module_copy = deepcopy(input_module, memo_tensors)
  File "/Users/jxiong/Documents/Projects/ipsc_gplvm/env/lib/python3.6/site-packages/gpflow/utilities/utilities.py", line 269, in deepcopy
    return copy.deepcopy(reset_cache_bijectors(input_module), memo)
  File "/Users/jxiong/Documents/Projects/ipsc_gplvm/env/lib/python3.6/site-packages/gpflow/utilities/utilities.py", line 253, in reset_cache_bijectors
    _ = traverse_module(input_module, accumulator, clear_bijector, target_types)
  File "/Users/jxiong/Documents/Projects/ipsc_gplvm/env/lib/python3.6/site-packages/gpflow/utilities/utilities.py", line 324, in traverse_module
    new_state = traverse_module(submodule, new_acc, update_cb, target_types)
  File "/Users/jxiong/Documents/Projects/ipsc_gplvm/env/lib/python3.6/site-packages/gpflow/utilities/utilities.py", line 324, in traverse_module
    new_state = traverse_module(submodule, new_acc, update_cb, target_types)
  File "/Users/jxiong/Documents/Projects/ipsc_gplvm/env/lib/python3.6/site-packages/gpflow/utilities/utilities.py", line 324, in traverse_module
    new_state = traverse_module(submodule, new_acc, update_cb, target_types)
  File "/Users/jxiong/Documents/Projects/ipsc_gplvm/env/lib/python3.6/site-packages/gpflow/utilities/utilities.py", line 303, in traverse_module
    return update_cb(m, path, state)
  File "/Users/jxiong/Documents/Projects/ipsc_gplvm/env/lib/python3.6/site-packages/gpflow/utilities/utilities.py", line 247, in clear_bijector
    clear_cache(bijector)
  File "/Users/jxiong/Documents/Projects/ipsc_gplvm/env/lib/python3.6/site-packages/gpflow/utilities/utilities.py", line 243, in clear_cache
    b._from_x.clear()
AttributeError: 'Softplus' object has no attribute '_from_x'
```

## Expected behavior

As shown in the example notebook(https://gpflow.readthedocs.io/en/master/notebooks/intro_to_gpflow2.html#Small-steps-big-changes), the function should clone the model and replace all gpflow.base.Parameters with tf.constants holding the same value.

## System information

* GPflow version:  2.0.5
* GPflow installed from:  pip install gpflow
* TensorFlow version:  2.3.0
* Python version   3.6.3 (v3.6.3:2c5fed86e0, Oct  3 2017, 00:32:08) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)]
* Operating system macOS Mojave Version 10.14.6
* Tensorflow_probability version: 0.11.0

## Additional context
The function also seems to fail on models other than SVGP.

**Comment by st--** [comment]
Hi @junhaobearxiong, tensorflow_probability changed the way how their bijector caches are implemented from version 0.10 (compatible with TensorFlow 2.2, incompatible with TensorFlow 2.3) to version 0.11 (compatible with 2.3, but no longer with 2.2). I _just_ made the changes to GPflow required to make freeze/deepcopy compatible with tensorflow_probability 0.11 and TensorFlow 2.3 (#1553), so this is already fixed - but these changes haven't been released yet. If you want to try out your code with TensorFlow 2.3, you can install GPflow from GitHub (`pip install git+https://github.com/GPflow/GPflow.git@develop#egg=gpflow`; if this complains about version incompatibilities during the install you may have to add `--use-feature=2020-resolver`). Otherwise we'll hopefully release this fix shortly as GPflow 2.1.1.

**Comment by junhaobearxiong** [comment]
Got it. Thanks for the help @st-- !
